tests/artifactoryupload_test.go

```diff
@@ -262,3 +246,53 @@ func explodeUpload(t *testing.T) {
 	}
 	artifactoryCleanup(t)
 }
+
+func propsUpload(t *testing.T) {
+	workingDir, _ := createWorkingDir(t)
+	defer os.RemoveAll(workingDir)
+
+	pattern := FixWinPath(filepath.Join(workingDir, "out", "*"))
+	up := services.NewUploadParams()
+	up.ArtifactoryCommonParams = &utils.ArtifactoryCommonParams{Pattern: pattern, Target: RtTargetRepo, AddProps: "key1=val1"}
+	up.Flat = true
+	uploaded, failed, err := testsUploadService.UploadFiles(up)
+	if uploaded != 1 {
+		t.Error("Expected to upload 1 file.")
+	}
+	if failed != 0 {
+		t.Error("Failed to upload", failed, "files.")
+	}
+	if err != nil {
+		t.Error(err)
+	}
+	searchParams := services.NewSearchParams()
+	searchParams.ArtifactoryCommonParams = &utils.ArtifactoryCommonParams{}
+	searchParams.Pattern = RtTargetRepo
+	searchParams.Props = "key1=val2"
```

Comment:
Are you sure this is correct? It looks like the value we're uploading is val1 and not val2, but I may be missing something.
If I am missing it, let's add some comments describing the test flow.

Reply:
Good catch! That test was broken.